server/src/main/java/org/apache/cassandra/sidecar/job/RepairJob.java

```diff
@@ -0,0 +1,249 @@
+        {
+            options.put(RepairOptions.COLUMNFAMILIES.getValue(), String.join(",", tables));
+        }
+
+        Boolean isPrimaryRange = repairPayload.isPrimaryRange();
+        if (isPrimaryRange != null)
+        {
+            options.put(RepairOptions.PRIMARY_RANGE.getValue(), String.valueOf(isPrimaryRange));
+        }
+        // TODO: Verify use-cases involving multiple DCs
+
+        String dc = repairPayload.datacenter();
+        if (dc != null)
+        {
+            options.put(RepairOptions.DATACENTERS.getValue(), dc);
+        }
+
+        List<String> hosts = repairPayload.hosts();
+        if (hosts != null && !hosts.isEmpty())
+        {
+            options.put(RepairOptions.HOSTS.getValue(), String.join(",", requireNonNull(hosts)));
+        }
+
+        if (repairPayload.startToken() != null && repairPayload.endToken() != null)
+        {
+            options.put(RepairOptions.RANGES.getValue(), repairPayload.startToken() + ":" + repairPayload.endToken());
+        }
+
+        if (repairPayload.repairType() == RepairPayload.RepairType.INCREMENTAL)
+        {
+            options.put(RepairOptions.INCREMENTAL.getValue(), Boolean.TRUE.toString());
+        }
+
+        if (repairPayload.force() != null)
+        {
+            options.put(RepairOptions.FORCE_REPAIR.getValue(), String.valueOf(repairPayload.force()));
+        }
+
+        if (repairPayload.isValidate() != null)
+        {
+            options.put(RepairOptions.PREVIEW.getValue(), PREVIEW_KIND_REPAIRED);
+        }
+        return options;
+    }
+
+    private void queryForCompletedRepair(Promise<Void> promise, int cmd)
+    {
+        LOGGER.info("Polling repair operation for jobId={} status={}", this.jobId(), promise.future().isComplete());
+        List<String> status = storageOperations.getParentRepairStatus(cmd);
+        String queriedString = "queried for parent session status and";
+        if (status == null)
+        {
+            LOGGER.error("{} couldn't find repair status for cmd: {}",
+                         queriedString, cmd);
+        }
+        else
+        {
+            ParentRepairStatus parentRepairStatus = ParentRepairStatus.valueOf(status.get(0));
+            LOGGER.info("Status from polling: {}", parentRepairStatus);
+            List<String> messages = status.subList(1, status.size());
+            switch (parentRepairStatus)
+            {
+                case COMPLETED:
+                case FAILED:
+                    LOGGER.info("{} discovered repair {}", queriedString, parentRepairStatus.name().toLowerCase());
+                    if (parentRepairStatus == ParentRepairStatus.FAILED)
+                    {
+                        promise.fail(new IOException(messages.get(0)));
+                    }
+                    LOGGER.info("Messages:\n{}", String.join("\n", messages));
+                    promise.tryComplete();
+                    vertx.cancelTimer(cmd);
```

Comment:
It should be timer ID, not the repair's cmd id right? I think timer needs to be cancelled whenever promise is resolved.